src/graphnet/utils/cluster/condor.py

```diff
@@ -0,0 +1,843 @@
+            returnString += "    Num jobs            : %i\n" % (
+                self.submit_fileInfo.numJobs
+            )
+        else:
+            returnString += "  [Submit file not parsed]\n"
+
+        if self.outputFileInfo != None:
+            returnString += "  Output file info :\n"
+            returnString += "    Return code          : %s\n" % (
+                str(self.outputFileInfo.returnCode)
+                if self.outputFileInfo.returnCode != None
+                else "[Not parsed]"
+            )
+        else:
+            returnString += "  [Output file not parsed]\n"
+
+        if self.metricsFileInfo != None:
+            returnString += "  Metrics file info :\n"
+            returnString += "    Num jobs             : %i\n" % (
+                self.metricsFileInfo.numJobs
+            )
+            returnString += "    Num jobs succeeded   : %i (%0.3f%%)\n" % (
+                self.metricsFileInfo.numJobsSucceeded,
+                percentage(
+                    self.metricsFileInfo.numJobsSucceeded,
+                    self.metricsFileInfo.numJobs,
+                ),
+            )
+            returnString += "    Num jobs failed      : %i (%0.3f%%)\n" % (
+                self.metricsFileInfo.numJobsFailed,
+                percentage(
+                    self.metricsFileInfo.numJobsFailed,
+                    self.metricsFileInfo.numJobs,
+                ),
+            )
+            returnString += "    Start time           : %s\n" % (
+                self.metricsFileInfo.start_time
+            )
+            returnString += "    End time             : %s\n" % (
+                self.metricsFileInfo.end_time
+            )
+            returnString += "    Time taken in total  : %s\n" % (
+                time.human_time(self.metricsFileInfo.duration)
+            )
+            returnString += "    DAG status           : %s\n" % (
+                interpretDAGStatus(self.metricsFileInfo.dag_status)
+            )
+        else:
+            returnString += "  [Metrics file not parsed]\n"
+
+        return returnString
+
+
+# Subclass containing submit file information
+class DAGManMetricsSubmitFileInfo(object):
+    def __init__(self, dagmanSubmitFilePath):
+        self.parse(dagmanSubmitFilePath)
+
+    def fullyParsed(self):
+        return True  # No conditional parsing for this file at this time
+
+    def parse(self, dagmanSubmitFilePath):
+
+        # Check submit exists, open it, and read it into a string
+        if os.path.exists(dagmanSubmitFilePath):
+            with open(dagmanSubmitFilePath) as submit_file:
+                submit_fileText = submit_file.read()
+
+                # Count the instances of the "JOB" ketword in the string
+                self.numJobs = submit_fileText.count("JOB")
+
+        else:
+            raise Exception(
+                'DAGMan submit file "%s" does not exist' % dagmanSubmitFilePath
+            )
```

Comment:
I don't see this class referenced elsewhere (apart from one or more places that are not themselves references elsewhere)?
```suggestion
```